### PR TITLE
Add Variant of resolve api which uses API KEY authentication

### PR DESCRIPTION
**Pull request by maxcbc**
- adds variant of resolve api which uses API KEY based authentication and a path param for contract id
 🐿 v2.10.2

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,7 +32,7 @@ provision:
 	nht float --master --no-destroy --skip-gtg --testapp ${TEST_APP} --vault
 
 run:
-	nht run --local --https
+	nht run --https
```
Comment (leggsimon) [inline Makefile]: I think you should be using `make run-local` instead so I'm not sure this should be removed.

File: server/middleware/get-contract-by-id-from-param.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+const getContractByID = require('../lib/get-contract-by-id');
+
+
+module.exports = (req, res, next) => {
+	const {contract_id} = req.params || {};
+	if (!contract_id) {
+		return res.sendStatus(400);
```
Comment (leggsimon) [inline server/middleware/get-contract-by-id-from-param.js]: Could this maybe send a message as well like, `contract_id not provided` or something like that?

File: server/middleware/get-contract-by-id-from-param.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	}
+	getContractByID(contract_id)
```
Comment (leggsimon) [inline server/middleware/get-contract-by-id-from-param.js]: does this need to be returned or awaited?

File: server/app.js [hunk 4]
```diff
@@ -113,8 +114,9 @@ if (process.env.NODE_ENV !== 'production') {
 		cache,
 		apiKey
 	];
-
+	app.post('/syndication/contracts/:contract_id/resolve', [...middleware, getContractByIdFromParam], require('./controllers/resolve'));
```
Comment (leggsimon) [inline server/app.js]: I don't think you'd necessarily need to use the array spread here. I think you could add the getContractByIdFromParam as the 3rd param but 🤷‍♂️ don't care really

```js
app.post(
	'/syndication/contracts/:contract_id/resolve',
	middleware,
	getContractByIdFromParam,
	require('./controllers/resolve')
);
```

File: Makefile (hunk 1 above)
Comment (maxcbc) [reply Makefile]: why?

File: server/app.js (hunk 4 above)
Comment (maxcbc) [reply server/app.js]: will ditch the spread

File: server/middleware/get-contract-by-id-from-param.js (hunk 3 above)
Comment (maxcbc) [reply server/middleware/get-contract-by-id-from-param.js]: Nope, not in a synchronous function.

File: server/app.js (hunk 4 above)
Comment (maxcbc) [reply server/app.js]: it is so ditched

File: server/middleware/get-contract-by-id-from-param.js (hunk 2 above)
Comment (maxcbc) [reply server/middleware/get-contract-by-id-from-param.js]: done and dusted

File: Makefile (hunk 1 above)
Comment (leggsimon) [reply Makefile]: IDK

Comment (leggsimon) [reply Makefile]: Cargo culting

File: server/middleware/get-contract-by-id-from-param.js (hunk 3 above)
Comment (leggsimon) [reply server/middleware/get-contract-by-id-from-param.js]: But `getContractByID` isn’t synchronous is it?

**Pull request merged**